sdk/inc/azure/iot/az_iot_hub_client.h

```diff
@@ -233,6 +233,19 @@ typedef struct
   } _internal;
 } az_iot_hub_client_properties;
 
+/**
+ * @brief Supported IoT Hub properties
+ */
+static const az_span az_iot_hub_client_properties_message_id
+    = AZ_SPAN_LITERAL_FROM_STR("%24.mid"); /**< Add unique identification to a message */
+static const az_span az_iot_hub_client_properties_correlation_id = AZ_SPAN_LITERAL_FROM_STR(
+    "%24.cid"); /**< Used in distributed tracing. More information here:
+                   https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-distributed-tracing */
+static const az_span az_iot_hub_client_properties_content_type = AZ_SPAN_LITERAL_FROM_STR(
+    "%24.ct"); /**< URL encoded and of the form text%2Fplain or application%2Fjson, etc */
+static const az_span az_iot_hub_client_properties_content_encoding
+    = AZ_SPAN_LITERAL_FROM_STR("%24.ce"); /**< utf-8, utf-16, etc */

```

Comment:
How are these values meant to be used?

For example, how is "%24.mid" unique?

Reply:
It is a "system property" recognized by IoT Hub. These are properties which we had available for users to set in the V1 SDK and we want to make them easily available to users.

Reply:
yes, these are system property that you can put on hub messages

Reply:
I want to understand these a bit more, hence the questions. Can you share some doc links that explain these or provide some sample snippets that showcase how the customer will use them?

For example, what are the implications of using `az_iot_hub_client_properties_content_encoding` when the JSON APIs expect UTF-8 only, and all our az_span strings are UTF-8 bytes? Similarly, why does using `%24.mid` make the id unique? Is the `%` in `%24` some sort of format string?

Reply:
The hub accepts UTF-8/UTF-16/UTF-32/avero/cbor message encodings.  The IoT SDK does not dictate the message payload, that is up to the app and the MQTT client. The message property names and values must be escaped when sent to the hub, so ''%24.mid" is actually "$.mid". All system properties start with a $.

https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-messages-construct

Reply:
> The message property names and values must be escaped when sent to the hub, so ''%24.mid" is actually "$.mid". All system properties start with a $.


I see, gotcha. Thanks for the explanation.

And this`$.mid` is just the **name** of the property, correct? And it's the **value** that will be user-provided that has to be unique?

Reply:
it's does not have to be unique. It could be anything they want. Usually used in routing the message to storage.

Reply:
> Add unique identification to a message

Then should we change the comment?